add http.{request,response}.body.content (#3521)
open-telemetry/semantic-conventions: docs/registry/attributes/http.md

| <a id="http-response-status-code" href="#http-response-status-code">`http.response.status_code`</a> | ![Stable](https://img.shields.io/badge/-stable-lightgreen) | int | [HTTP response status code](https://tools.ietf.org/html/rfc7231#section-6). | `200` |
| <a id="http-route" href="#http-route">`http.route`</a> | ![Stable](https://img.shields.io/badge/-stable-lightgreen) | string | The matched route template for the request. This MUST be low-cardinality and include all static path segments, with dynamic path segments represented with placeholders. [7] | `/users/:userID?`; `my-controller/my-action/{id?}` |

**[1] `http.request.body.content`:** Captured value MAY be limited in size and thus value is expected to be truncated in many cases.

The maximum captured body size must be limited by the `AttributeValueLengthLimit` configuration option defined in the

> [OpenTelemetry SDK specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/README.md#configurable-parameters).

Reviewer: This is not actionable for instrumentation authors  - let's remove ?

Author: removed in 7d5fedf1

> To prevent excessive overhead and storage by default, instrumentations MUST issue a warning when the feature is used without an explicit `AttributeValueLengthLimit` configuration option.

Reviewer: instrumentation has no knowledge of configuration applied by the application.

I'd propose to just remove this paragraph

Author: From what I understand this means that without access to the SDK configuration, we have no way to issue a warning when the feature is used without limit (which is something we may want to avoid).

Do you think it would make sense to add the following requirement for feature implementation documentation ?

> Enabling this feature is likely to capture large values and thus the feature documentation MUST recommend to set an explicit value for `AttributeValueLengthLimit` to avoid excessive overhead ?

Author: removed in 7d5fedf1

> Instrumentations MUST disable capture by default and provide an opt-in configuration option to enable it as it can capture sensitive information and cause performance overhead.

Reviewer: ```suggestion
Instrumentations MUST NOT capture this attribute by default and MAY provide an option to enable it.

> [!WARNING]
> This attribute may contain sensitive information.
```

a bit more clarity and also this is a standard incantation we use in semconv to mark sensitive things.

Reviewer: it seems it was lost since it was applied to md and regenerated from yaml, adding comment to yaml, please make sure to update response body too

Author: fixed in 34d61ac6

Instrumentations SHOULD capture the body as string whenever possible as it makes it easier to use in human-readable form, also it allows to implement sanitization if needed.

Capturing HTTP request body content should be considered as fragile by nature as it depends on implementation details,

> configuration and usage of the instrumented HTTP client and server libraries.

Reviewer: Let's move this paragraph somewhere so it does not separate parts about string / text content that are currently before and after it.

Also, I'd prefer to rephrase it to something more specific like

```
Instrumentations that implement request body recording MUST NOT intentionally introduce side effects such as changing stream position or closing body stream independently from the application.

When body is recorded on HTTP client or server spans, the instrumentation SHOULD record part of the body that was sent or received at the time HTTP span has ended. 
```

Also applicable to response body
